src/Database/Pgsql/Driver/Pgsql.php

```diff
@@ -0,0 +1,135 @@
+
+        return $config;
+    }
+
+    public function create(Tenant $tenant): bool
+    {
+        $config = $this->configure($tenant);
+
+        event(new Events\Creating($tenant, $config, $this));
+
+        $connection = $this->system($tenant);
+
+        $result = $this->queryManager->setConnection($connection)
+                     ->process(function () use ($config) {
+                         $this->statement("CREATE USER \"{$config['username']}\" WITH PASSWORD '{$config['password']}'");
+                         if ($config['schema'] == 'public') {
+                             $this->statement("CREATE DATABASE \"{$config['database']}\" WITH OWNER = \"{$config['username']}\"");
+                         } else {
+                             $this->statement("CREATE SCHEMA IF NOT EXISTS \"{$config['schema']}\" AUTHORIZATION \"{$config['username']}\"");
+                         }
+                     })
+                     ->getStatus();
+
+        event(new Events\Created($tenant, $this, $result));
+
+        return $result;
+    }
+
+    public function update(Tenant $tenant): bool
+    {
+        $config = $this->configure($tenant);
+
+        event(new Events\Updating($tenant, $config, $this));
+
+        if (!isset($config['oldUsername'])) {
+            return false;
+        }
+
+        $connection = $this->system($tenant);
+
+        $result = $this->queryManager->setConnection($connection)
+                     ->processTransaction(function () use ($config) {
+                         $this->statement("ALTER USER \"{$config['oldUsername']}\" RENAME TO \"{$config['username']}\"");
+                         $this->statement("ALTER USER \"{$config['username']}\" WITH PASSWORD '{$config['password']}'");
+                     })
+                     ->process(function () use ($config) {
+                         if ($config['schema'] == 'public') {
+                             $this->statement("ALTER DATABASE \"{$config['oldUsername']}\" RENAME TO \"{$config['database']}\"");
+                             $this->statement("ALTER DATABASE \"{$config['database']}\" OWNER TO \"{$config['username']}\"");
+                         } else {
+                             $this->statement("ALTER SCHEMA \"{$config['oldUsername']}\" RENAME TO \"{$config['schema']}\"");
+                             $this->statement("ALTER SCHEMA \"{$config['schema']}\" OWNER TO \"{$config['username']}\"");
+                         }
+                     })
+                     ->getStatus();
+
+        event(new Events\Updated($tenant, $this, $result));
+
+        return $result;
+    }
+
+    public function delete(Tenant $tenant): bool
+    {
+        $config = $this->configure($tenant);
+
+        event(new Events\Deleting($tenant, $config, $this));
+
+        $result = $this->queryManager->setConnection($this->system($tenant))
+            ->process(function () use ($config) {
+                if ($config['schema'] == 'public') {
```

Comment:
Here I find it hard to assume that with anything other than public, we shouls remove the schema.... Dont think users will see it coming 😅

Reply:
quite possibly... `public` is the default schema name for a database in postgres, so if you're leaving the config alone and simply changing the database name (which I believe 99.999% of database-separation users will do), then it will certainly be public.

Any suggestions for how we should do it then?

Reply:
This is the only thing I worry about as well especially not knowing pgsql.
What happens if someone does change the default schema name for what ever reason? Have we tested that, and if we have 2 tenants would it delete both of their databases? (again not 100% on pgsql)

Reply:
I don't think you CAN change the default name (nor can I think of a rational reason you'd want to).  But you can certainly have a separate schema as your default "system" one, and that's what it would see.

I'm open to suggestions though

Reply:
Changing the default schema is possible in PSQL (see [external resource](https://mkyong.com/database/postgresql-how-to-change-default-schema/))

Reply:
Why not allow a static setter on the Pgsql class to set the default schema? And then retrieve that in this conditional?

Reply:
public schema gets created on every new postgresql database created.

The default template1 is copied to the new db on create. By default, every database created contains a public schema. This is for the working data, which us developers usually work with. A database might contain a lot more schemas, some by default from Postgresql, some from extensions like Postgis.

If config has it enabled, the database should ve deleted, not the public schema.